core/runtime/bindings.rs

```diff
@@ -356,6 +356,34 @@ pub extern "C" fn host_initialize_import_meta_object_callback(
   let url_val = v8::String::new(scope, &name).unwrap();
   meta.create_data_property(scope, url_key.into(), url_val.into());
 
+  // For `file:` URL we provide additional `filename` and `dirname` values
+  if name.starts_with("file:") {
+    let filename_key =
+      v8::String::new_external_onebyte_static(scope, b"filename").unwrap();
+    let dirname_key =
+      v8::String::new_external_onebyte_static(scope, b"dirname").unwrap();
+    let filename_val =
+      v8::String::new(scope, &name.strip_prefix("file://").unwrap()).unwrap();
+    let dirname = name
+      .strip_prefix("file://")
+      .unwrap()
+      .rsplit_once("/")
+      .map(
+        |(dirname, _)| {
+          if dirname.is_empty() {
+            "/"
+          } else {
+            dirname
```

Comment:
We need to add windows support here

Reply:
I believe this is unreachable on Windows? There's gonna be something like `file://C:/foo/bar/something.js`

Reply:
Sorry, I just meant in general (unless I'm misreading this). For example, something like `file:///C:/foo/bar/something.js` (note three forward slashes) would give `C:\foo\bar\something.js` similar to in Node.

I believe the output should be the same as:

```ts
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
```

Reply:
I see - so you want to change the file URL to be an actual path (with backwards slashes). Do you happen to know if we already have this utility somewhere in `deno`?

Reply:
Yeah, I think we'd need to parse it to a url then use `to_file_path()`

Just tried this out to confirm in Node:

```shellsession
> cat test.mjs
console.log(import.meta.filename);
console.log(import.meta.dirname);
> node test.mjs
V:\scratch\test.mjs
V:\scratch
```

Reply:
Thanks, I'll fix it in a moment

Reply:
That said, I kind of wonder if we could just do our own thing here without using the url crate.... might not be too hard and it would save us having to parse the file url, which would be faster. https://github.com/servo/rust-url/blob/f447500049cbfe42203ff16026842e74f3fd99ac/url/src/lib.rs#L2598-L2611

Reply:
I think it's not a big deal, this is called only once per module.